lib/payment.js

```diff
@@ -154,7 +155,8 @@ class Payment {
 
         const result = Object.assign({}, amount, {
             sender: {
-                wallet: _sender.get(this)
+                wallet: _sender.get(this),
+                data: _senderData.get(this)
```

Comment:
I think this is a better place to enforce the data to be an empty string in case `_senderData.get(this)` is "falsy".